### PR TITLE
Refresh depends for all r pkgs

### PR DESCRIPTION
**Pull request by sukanka**
## Involved packages

This pr refreshes `depends` and `optdepends` for all r pkgs in our repo.
This pr also tends to unarchive the archived packages in our repo which have later been available again in CRAN (e.g. `r-bayesrel`) 

now ready for merge
## Involved issue

Close #

## How to review
just have a look at the unarchived or archived pkgs. 

pkgs with `dependes` and `optdends` changes can be safely ignored.

## Details

- [ ] Tested in the local machine (largest 16G RAM) and it is passed without any issue
- [ ] Provide New Package
- [ ] Fix the Packages
  - [ ] PKGBUILD
  - [ ] lilac.yaml
  - [ ] lilac.py
- [ ] Would like to continue to work with us

### CODE REVIEW

File: BioArchLinux/r-alevinqc/PKGBUILD [hunk 1]
```diff
@@ -16,6 +16,7 @@ depends=(
   r-dt
   r-ggally
   r-ggplot2
+  r-rcpp
```
Comment (sukanka) [inline BioArchLinux/r-alevinqc/PKGBUILD:19]: this breaks `arch=('any')`, now `arch=('x86_64`)`

File: BioArchLinux/r-arrow/PKGBUILD [hunk 2]
```diff
@@ -32,16 +35,17 @@ optdepends=(
...
   r-rmarkdown
   r-stringi
   r-stringr
+  r-sys
   r-testthat
   r-tibble
+  r-tzdb
   r-withr
-  libcurl-compat
-  openssl
```
Comment (sukanka) [inline BioArchLinux/r-arrow/PKGBUILD:44]: @starsareintherose  why does this pkg optionally depends on `libcurl-compat` and `openssl`?

File: BioArchLinux/r-igraph/PKGBUILD [hunk 3]
```diff
@@ -27,9 +27,6 @@ optdepends=(
   r-tcltk
   r-testthat
   r-withr
-  gmp
-  glpk
-  libxml2
 )
```
Comment (sukanka) [inline BioArchLinux/r-igraph/PKGBUILD:30]: @starsareintherose  please add description to these optional depends

File: BioArchLinux/r-phemd/PKGBUILD [hunk 4]
```diff
@@ -4,10 +4,10 @@ _pkgname=phemd
 _pkgver=1.14.0
 pkgname=r-${_pkgname,,}
 pkgver=1.14.0
-pkgrel=1
+pkgrel=2
 pkgdesc='Phenotypic EMD for comparison of single-cell samples'
 arch=('any')
-url="https://bioconductor.org/packages/${_pkgname}"
+url="https://bioconductor.org/packages/3.15/${_pkgname}"
```
Comment (sukanka) [inline BioArchLinux/r-phemd/PKGBUILD:10]: this is not archived

File: BioArchLinux/r-pkgmaker/lilac.yaml [hunk 5]
```diff
@@ -12,5 +12,5 @@ repo_depends:
 update_on:
 - regex: pkgmaker_([\d._-]+).tar.gz
   source: regex
-  url: https://cran.r-project.org/package=pkgmaker
+  url: https://cran.r-project.org/src/contrib/Archive/package=pkgmaker
```
Comment (sukanka) [inline BioArchLinux/r-pkgmaker/lilac.yaml:15]: this url is wrong

File: BioArchLinux/r-arrow/PKGBUILD (hunk 2 above)
Comment (starsareintherose) [reply BioArchLinux/r-arrow/PKGBUILD:44]: I forget, you can check

File: BioArchLinux/r-igraph/PKGBUILD (hunk 3 above)
Comment (starsareintherose) [reply BioArchLinux/r-igraph/PKGBUILD:30]: this is generated by python script, commit on pkgbuild-generator

File: BioArchLinux/r-arrow/PKGBUILD (hunk 2 above)
Comment (sukanka) [reply BioArchLinux/r-arrow/PKGBUILD:44]: > I forget, you can check

I cannot check, see https://github.com/BioArchLinux/Packages/commit/608ef961a51c71a6ca1451738e230ee46a5f23fa

File: BioArchLinux/r-igraph/PKGBUILD (hunk 3 above)
Comment (sukanka) [reply BioArchLinux/r-igraph/PKGBUILD:30]: > this is generated by python script, commit on pkgbuild-generator

nope, they are introduced by you, see https://github.com/BioArchLinux/Packages/commit/082fb4cb090ec3ee2865d5ec2ff9b16e0b8c183f

Comment (sukanka) [reply BioArchLinux/r-igraph/PKGBUILD:30]: Oh I see, they are in the top of the PKGBUILD and marked as optional

File: BioArchLinux/r-alps/PKGBUILD [hunk 6]
```diff
@@ -4,10 +4,10 @@ _pkgname=ALPS
 _pkgver=1.8.0
 pkgname=r-${_pkgname,,}
 pkgver=1.8.0
-pkgrel=4
+pkgrel=5
 pkgdesc='AnaLysis routines for ePigenomicS data'
 arch=('any')
-url="https://bioconductor.org/packages/${_pkgname}"
+url="https://bioconductor.org/packages/3.13/${_pkgname}"
```
Comment (sukanka) [inline BioArchLinux/r-alps/PKGBUILD:10]: can be changed to 3.14

Comment (starsareintherose) [reply BioArchLinux/r-alps/PKGBUILD:10]: okay

File: BioArchLinux/r-phemd/PKGBUILD (hunk 4 above)
Comment (starsareintherose) [reply BioArchLinux/r-phemd/PKGBUILD:10]: re-use it

File: BioArchLinux/r-pzfx/lilac.yaml [hunk 7]
```diff
@@ -5,10 +5,7 @@ maintainers:
 update_on:
 - regex: pzfx_([\d._-]+).tar.gz
   source: regex
-  url: https://cran.r-project.org/src/contrib 
+  url: https://cran.r-project.org/package=contrib
```
Comment (sukanka) [inline BioArchLinux/r-pzfx/lilac.yaml:8]: This line is wrong, I should manually fix it

File: BioArchLinux/r-reffreeewas/PKGBUILD [hunk 8]
```diff
@@ -4,10 +4,10 @@ _pkgname=RefFreeEWAS
 _pkgver=2.2
 pkgname=r-${_pkgname,,}
 pkgver=2.2
-pkgrel=8
+pkgrel=9
 pkgdesc=' EWAS using Reference-Free DNA Methylation Mixture Deconvolution'
 arch=('any')
-url="https://bioconductor.org/packages/${_pkgname}"
+url="https://bioconductor.org/packages/None/${_pkgname}"
```
Comment (sukanka) [inline BioArchLinux/r-reffreeewas/PKGBUILD:10]: This line is wrong, manually fix it

**Pull request merged**